modules/webhook/feishu.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package webhook
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/git"
+	api "code.gitea.io/gitea/modules/structs"
+)
+
+type (
+	// FeishuPayload represents
+	FeishuPayload struct {
+		Title string `json:"title"`
+		Text  string `json:"text"`
+	}
+)
+
+// SetSecret sets the Feishu secret
+func (p *FeishuPayload) SetSecret(_ string) {}
+
+// JSONPayload Marshals the FeishuPayload to json
+func (p *FeishuPayload) JSONPayload() ([]byte, error) {
+	data, err := json.MarshalIndent(p, "", "  ")
+	if err != nil {
+		return []byte{}, err
+	}
+	return data, nil
+}
+
+func getFeishuCreatePayload(p *api.CreatePayload) (*FeishuPayload, error) {
+	// created tag/branch
+	refName := git.RefEndName(p.Ref)
+	title := fmt.Sprintf("[%s] %s %s created", p.Repo.FullName, p.RefType, refName)
+
+	return &FeishuPayload{
+		Text:  title,
+		Title: title,
+	}, nil
+}
+
+func getFeishuDeletePayload(p *api.DeletePayload) (*FeishuPayload, error) {
+	// created tag/branch
+	refName := git.RefEndName(p.Ref)
+	title := fmt.Sprintf("[%s] %s %s deleted", p.Repo.FullName, p.RefType, refName)
+
+	return &FeishuPayload{
+		Text:  title,
+		Title: title,
+	}, nil
+}
+
+func getFeishuForkPayload(p *api.ForkPayload) (*FeishuPayload, error) {
+	title := fmt.Sprintf("%s is forked to %s", p.Forkee.FullName, p.Repo.FullName)
+
+	return &FeishuPayload{
+		Text:  title,
+		Title: title,
+	}, nil
+}
+
+func getFeishuPushPayload(p *api.PushPayload) (*FeishuPayload, error) {
+	var (
+		branchName = git.RefEndName(p.Ref)
+		commitDesc string
+	)
+
+	title := fmt.Sprintf("[%s:%s] %s", p.Repo.FullName, branchName, commitDesc)
+
+	var text string
+	// for each commit, generate attachment text
+	for i, commit := range p.Commits {
+		var authorName string
+		if commit.Author != nil {
+			authorName = " - " + commit.Author.Name
+		}
+		text += fmt.Sprintf("[%s](%s) %s", commit.ID[:7], commit.URL,
+			strings.TrimRight(commit.Message, "\r\n")) + authorName
```

Comment:
```suggestion
		text += fmt.Sprintf("[%s](%s) %s - %s", commit.ID[:7], commit.URL,
			strings.TrimRight(commit.Message, "\r\n"), commit.Author.Name)
```
It think otherwise the author name will be united to the message (e.g. `This is my commitguillep2k` vs. `This is my commit - guillep2k`)

Reply:
I don't think so. `authorName` will be blank or `" - " + commit.Author.Name`. So it will not.

Reply:
Sorry, you're right, I see it now. I was just comparing side-by-side with `msteams.go` and got confused.